types/stylelint/index.d.ts

```diff
@@ -53,13 +53,15 @@ export namespace formatters {
 
 export function lint(options?: LinterOptions): Promise<LinterResult>;
 
-export type RuleOption = {
-    actual: any;
-    possible?: any;
+export type RuleOption = any;
+
+export type RuleValidationOption = {
```

Comment:
Please revert all this. 

Also it would be nice, to keep the same namings like in stylelint and not come up with custom names, as it would be confusing, apart from that creating a custom type for `any` is redundant.

In stylelint this is refered to as `OptionDescriptions`

Also looking at the previous PR, it looks like the `createPlugin` is not implemented fully, as it can accept two parameters https://stylelint.io/developer-guide/plugins/

stylelint.c

> CreatePlugin(ruleName, ruleFunction) ensures that your plugin will be setup properly alongside other rules.
> 
> In order for your plugin rule to work with the standard configuration format, ruleFunction should accept 2 arguments: the primary option and, optionally, a secondary options object.
> 

Apart from that it doesn't look like it's supposed to be an array.

Thus, IMHO the correct change would be just to update the `createPlugin` to 
```

export function createPlugin(
    ruleName: string,
    plugin: (primaryOption: RuleOption, secondaryOption?: RuleOption) => (root: any, result: LintResult) => void | Promise<void>,
):  void;

```